Modules/Core/Common/include/itkImageSink.hxx

```diff
@@ -0,0 +1,287 @@
+                                       inputImageRegion );
+  m_CurrentInputRegion = inputImageRegion;
+
+  itkDebugMacro( "Generating " << inputRequestedRegionNumber << " chunk as " <<  m_CurrentInputRegion );
+
+
+  for (auto &inputName: this->GetInputNames())
+    {
+    if ( this->ProcessObject::GetInput(inputName) )
+      {
+      // Check whether the input is an image of the appropriate
+      // dimension (use ProcessObject's version of the GetInput()
+      // method sinke it returns the input as a pointer to a
+      // DataObject as opposed to the subclass version which
+      // static_casts the input to an TInputImage).
+      using ImageBaseType = ImageBase< InputImageDimension >;
+      typename ImageBaseType::ConstPointer constInput =
+        dynamic_cast< ImageBaseType const * >( this->GetInput(inputName) );
+
+      // If not an image, skip it, and let a subclass of
+      // ImageToImageFilter handle this input.
+      if ( constInput.IsNull() )
+        {
+        continue;
+        }
+
+      // Input is an image, cast away the constness so we can set
+      // the requested region.
+      InputImagePointer input = const_cast< TInputImage * >( this->GetInput(inputName) );
+
+      // copy the requested region of the first input to the others
+      InputImageRegionType inputRegion;
+      input->SetRequestedRegion( m_CurrentInputRegion );
+      }
+    }
+}
+
+
+template <class TInputImage>
+void
+ImageSink<TInputImage>
+::VerifyInputInformation() ITKv5_CONST
+{
+  using ImageBaseType = const ImageBase< InputImageDimension >;
+
+  ImageBaseType *inputPtr1 = nullptr;
+  InputDataObjectConstIterator it(this);
+
+  for(; !it.IsAtEnd(); ++it )
+    {
+    // Check whether the output is an image of the appropriate
+    // dimension (use ProcessObject's version of the GetInput()
+    // method since it returns the input as a pointer to a
+    // DataObject as opposed to the subclass version which
+    // static_casts the input to an TInputImage).
+    inputPtr1 = dynamic_cast< ImageBaseType * >( it.GetInput() );
+
+    if ( inputPtr1 )
+      {
+      break;
+      }
+    }
+
+    for (; !it.IsAtEnd(); ++it )
+    {
+    auto * inputPtrN = dynamic_cast< ImageBaseType * >( it.GetInput() );
+
+    // Physical space computation only matters if we're using two
+    // images, and not an image and a constant.
+    if( inputPtrN )
+      {
+      // check that the image occupy the same physical space, and that
+      // each index is at the same physical location
+
+      // tolerance for origin and spacing depends on the size of pixel
+      // tolerance for directions a fraction of the unit cube.
+      const SpacePrecisionType coordinateTol
+        = std::abs(this->m_CoordinateTolerance * inputPtr1->GetSpacing()[0]); // use first dimension spacing
+
+      if ( !inputPtr1->GetOrigin().GetVnlVector().is_equal(inputPtrN->GetOrigin().GetVnlVector(), coordinateTol) ||
```

Comment:
Not sure about this, but I thought there was an attempt to avoid usage of vnl structures so that at some point in the future we can move away from vnl. Is there some other way of doing the comparisons that isn't vnl dependent? I may be wrong about this.

Reply:
This is the best method available. Same as imagetoimage.